pull-requests.md

```diff
@@ -0,0 +1,162 @@
+
+If you must submit a large PR, try to at least make someone else aware of this fact, and arrange for their time to review and get the PR merged. It's not fair to the team to dump large pieces of work on their laps without warning.
+
+If you can break up your changes into multiple pull requests then do that.
+
+
+
+
+
+Reviewing Pull Requests
+-----
+
+It's a reviewers responsibility to ensure:
+
+* Commit history is excellent
+* Good changes are propagated quickly
+* Code review is performed
+* Anti patterns are flagged
+* Out of scope changes are omitted
+* They understand what is being changed, from the perspective of someone examining the code in the future.
+
+### Reviewers are the guardians of the commit history
+
+The importance of ensuring a quality commit history cannot be stressed enough. It is the historical context of all of the work that we do, and is vital for understanding the reasons why changes were made in the past. What is obvious now, will not be obvious 2-3 years in the future.
+
+Without a decent commit history, we may as well be storing all our code in files ending yyyy-mm-dd. The commit history of a code base is what allows people to understand **why** a change was made - the when, what, and where are automatically evident.
+
+When looking at a commit message, ask yourself the question - from the perspective of someone looking at this change without any knowledge of the codebase - 'do I understand *why* this change was made?'
+
+
+
+### Keep the flow going
+
+Pull Requests are the fundamental unit of how we progress change. If PRs are getting clogged up in the system, pending review, they are preventing a piece of work from being completed.
+
+As PRs clog up in the system, merges become more difficult, as other features and fixes are applied to the same codebase. This in turn slows them down further, and often completely blocks progress on a given codebase.
+
+There is a balance between flow and ensuring the quality of our PRs. As a reviewer you should make a call as to whether a code quality issue is sufficient enough to block the PR.
+
+Any quality issue that will obviously result in a bug should be fixed.
+
+
+### We are all reviewers
+
+To make sure PRs flow through the system speedily, we must scale the PR review process. It is not sufficient (or fair!) to expect one or two people to review all PRs to our code. For starters, it creates a blocker every time those people are busy.
+
+Hopefully with the above guidelines, we can all start sharing the responsibility of being a reviewer.
+
+NB: With this in mind - if you are the first to comment on a PR, you are that PRs reviewer. If you feel that you can no longer be responsible for the subsequent merge or closure of the PR, then flag this up in the PR conversation, so someone else can take up the role.
+
+There's no reason why multiple people cannot comment on a PR and review it, and this is to be encouraged.
+
+
+### Don't add to the PR yourself.
```

Comment:
Nope - if you add to the PR you are no longer a reviewer (which is oke, as long as you are aware of this)

Reply:
Addressed in 09378360c9d5431828344774ac8ab22cdd039a4d